pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -304,6 +304,99 @@ type MonitoringSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +optional
 	ExternalMgrPrometheusPort uint16 `json:"externalMgrPrometheusPort,omitempty"`
+	// AlertRuleOverrides points to a customized Ceph prometheus alerts
+	// +optional
+	// +nullable
+	AlertRuleOverrides *Alerts `json:"alertRuleOverrides,omitempty"`
```

Comment:
I wonder if we should keep the CRD simpler by using a more general definition for the overrides. Talking to @pcuzner there are many more alert rules being created by the ceph team, and we are soon going to have over 50 alerts. If they are all individually listed in the CRD it will be a burden to maintain the long list and update it every time Ceph adds a new alert. If we define a single `CephAlert` struct with all the properties that can be overridden, then we can keep these overrides very simple, and no need to update the CRD when Ceph updates its rules. We do lose stronger type checks from the CRD validation, but I would vote for simplicity over the type checks in this case because there would be so many to maintain. @yuvalman wdyt?
```suggestion
	AlertRuleOverrides map[string]CephAlert `json:"alertRuleOverrides,omitempty"`
```

Reply:
I agree, it will be much easier to maintain the CRD that way.
I'll update the pr accordingly.